pkg/utils/transform/transform.go

```diff
@@ -48,3 +49,54 @@ func InjectNamespaceConditional(preserveNamespace string, targetNamespace string
 		return tf(u)
 	}
 }
+
+func ReplaceKind(fromKind, toKind string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+
+		if kind != fromKind {
+			return nil
+		}
+		err := unstructured.SetNestedField(u.Object, toKind, "kind")
+		if err != nil {
+			return fmt.Errorf("could change resource KIND, %q", err)
+		}
+		return nil
+	}
+}
+
+func InjectLabel(key, value string, overwrite bool, kinds ...string) mf.Transformer {
```

Comment:
can this be unit-tested?

Reply:
or even better have `kind string, rest ...string` so that user is forced to pass a kind atleast.

Reply:
yes, i am adding tests.

by keeping `kind` variadic, if we don't pass any kind the transformer will add labels to all resources.